public/app/plugins/panel/geomap/editor/StyleRuleEditor.tsx

```diff
@@ -86,30 +111,36 @@ export const StyleRuleEditor: FC<StandardEditorProps<FeatureStyleConfig, any, an
     <div className={styles.rule}>
       <InlineFieldRow className={styles.row}>
         <InlineField label="Rule" labelWidth={LABEL_WIDTH} grow={true}>
-          <Input
-            type="text"
+          <Select
+            menuShouldPortal
             placeholder={'Feature property'}
             value={`${value?.rule?.property}`}
-            onChange={onChangeComparisonProperty}
+            options={propertyOptions}
+            onChange={onChangeProperty}
             aria-label={'Feature property'}
+            isClearable={true}
+            allowCustomValue={true}
```

Comment:
Side note, for bool props we can just pass the prop which is the same thing as setting it to `true` (i.e. just pass `isClearable` and `allowCustomValue` by themselves